### PR TITLE
Barebones node server

### PR DESCRIPTION
**Pull request by JonathanRosado**

### CODE REVIEW

File: app/routes/test.es6 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+let index = (req, res) => {
+    res.json({hey: 'hello'});
+};
+
+exports.index = index;
```
Comment (rqc) [inline app/routes/test.es6]: @JonathanRosado could you please add newline characters at the end of the files flagged?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    "mocha": "^3.0.2",
+    "morgan": "~1.7.0",
+    "nconf": "^0.8.4",
+    "nsp": "^2.6.1",
+    "supertest": "^2.0.0",
+    "winston": "~2.2.0"
+  },
+  "devDependencies": {},
+  "scripts": {
+    "test": "nsp check && mocha test/tests.es6",
```
Comment (rqc) [inline package.json]: `nsp` awesome!

File: server.es6 [hunk 3]
```diff
@@ -0,0 +1,43 @@
+let express = require('express');
```
Comment (rqc) [inline server.es6]: I think you can add multiple key / value pairs to a *single* `let` statement.  https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

File: server.es6 [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+app.use(helmet());
+
+// Parse application/json
+app.use(bodyParser.json());
+
+app.get('/', testRoute.index);
+
+let privateKey;
+let certificate;
+let passphrase;
```
Comment (rqc) [inline server.es6]: merge `let` statements into one.

File: server.es6 [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+try {
+    privateKey  = fs.readFileSync('key.pem', 'utf8');
+    certificate = fs.readFileSync('cert.pem', 'utf8');
+    passphrase ='default'
+} catch (err) {
+    console.log("An error occurred while search for `key.pem` and `cert.pem`. " + err.toString())
```
Comment (rqc) [inline server.es6]: does this `console.log()` call default to `logger` (`morgan`)?

File: server.es6 [hunk 6]
```diff
@@ -0,0 +1,43 @@
+let express = require('express');
+let https = require('https');
+let fs = require('fs');
+let config = require('nconf');
+let bodyParser = require('body-parser');
+let morgan = require('morgan');
```
Comment (rqc) [inline server.es6]: I see that `morgan` is more of an `http` logger and `winston` is a generic `logger`. How do these two work together? Our next phase is to add `logging` aggregation for auditing, so working out the kinks now might come a long way. Thanks!

File: server.es6 [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+let logger = require('winston');
+var helmet = require('helmet');
+
+let testRoute = require('./app/routes/test.es6');
+
+// Load express
+let app = express();
+
+// Helmet can help protect your app from some well-known web vulnerabilities by setting HTTP headers appropriately.
```
Comment (rqc) [inline server.es6]: Nice! Get this kid a `helmet`!

File: server.es6 [hunk 8]
```diff
@@ -0,0 +1,43 @@
+let express = require('express');
+let https = require('https');
+let fs = require('fs');
+let config = require('nconf');
+let bodyParser = require('body-parser');
```
Comment (rqc) [inline server.es6]: is `body-parser` better?

File: app/routes/test.es6 [hunk 9]
```diff
@@ -0,0 +1,5 @@
+let index = (req, res) => {
```
Comment (rqc) [inline app/routes/test.es6]: what is the advantage of using `let` here, instead of `var`?

**Comment by rqc** [comment]
@JonathanRosado it is looking good. Implement those minor tweaks and I will merge. Great job!!

File: README.md [hunk 10]
```diff
@@ -1 +1,4 @@
 # he-auth-service
+
+# Generate key cert
+openssl req -x509 -newkey rsa:4096 -keyout key.pem -out cert.pem -days 9999
```
Comment (rqc) [inline README.md]: add back ticks to make it prettier :wink:

File: server.es6 (hunk 3 above)
Comment (rqc) [inline server.es6]: I changed my mind on this. Even if it is more verbose, I think it will be easier to read.

File: app/routes/test.es6 (hunk 1 above)
Comment (JonathanRosado) [inline app/routes/test.es6]: @rqc sure

File: app/routes/test.es6 (hunk 9 above)
Comment (JonathanRosado) [inline app/routes/test.es6]: The scoping of `let` is more intuitive to me. There's no real advantage here, the idea is to get in the habit of always favoring `let`.

File: README.md (hunk 10 above)
Comment (JonathanRosado) [inline README.md]: @rqc sure, ill make a proper readme as i go. Im just filling with bullet points

File: server.es6 [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+app.get('/', testRoute.index);
+
+let privateKey;
+let certificate;
+let passphrase;
+
+try {
+    privateKey  = fs.readFileSync('key.pem', 'utf8');
+    certificate = fs.readFileSync('cert.pem', 'utf8');
+    passphrase ='default'
```
Comment (JonathanRosado) [inline server.es6]: "default" string value. I think a password is needed when creating an ssl cert (have to double check). 
I've given it "default" for now.

File: server.es6 (hunk 5 above)
Comment (JonathanRosado) [inline server.es6]: @rqc need to set up morgan and winston

File: server.es6 (hunk 6 above)
Comment (JonathanRosado) [inline server.es6]: @rqc Yeah, the idea is to use morgan middleware to log http interactions. Winston would be used for logging per se.

**Comment by rqc** [comment]
@JonathanRosado I am merging this, please make any tweaks on a separate pull request. Thanks!

**Pull request merged**